src/Sparrow/LowMemory/MemoryInformation.cs

```diff
@@ -77,73 +78,98 @@ public struct MemoryStatusEx
             // getting it from /proc/self/stat or statm can be also problematic because in some distros the number is in page size, in other pages, and position is not always guarenteed
             // however /proc/self/status gives the real number in humenly format. We extract this here:
-            var filterString = KernelVirtualFileSystemUtils.ReadLineFromFile(path, filter);
-            if (filterString == null)
-            {
-                if (Logger.IsInfoEnabled)
-                    Logger.Info($"Failed to read {filter} from {path}");
-                return 0;
-            }
-
-            var parsedLine = filterString.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
 
-            if (parsedLine.Length != 3) // format should be: {filter}: <num> kb
+            var rcValues = new long[filters.Length];
+            
+            var filtersString = KernelVirtualFileSystemUtils.ReadLineFromFile(path, filters);
+            if (filtersString == null)
             {
                 if (Logger.IsInfoEnabled)
-                    Logger.Info($"Failed to read {filter} from {path}. Line was {parsedLine}");
-                return 0;
+                    Logger.Info($"Failed to read " + string.Join(",", filters) + " from {path}");
+                return null;
             }
 
-            if (parsedLine[0].Contains($"{filter}:") == false)
+            var cnt = 0;
+            foreach (var filterString in filtersString)
             {
-                if (Logger.IsInfoEnabled)
-                    Logger.Info($"Failed to find {filter} from {path}. Line was {parsedLine}");
-                return 0;
-            }
+                var parsedLine = filterString.Split(new[] {' '}, StringSplitOptions.RemoveEmptyEntries);
 
-            if (long.TryParse(parsedLine[1], out var result) == false)
-            {
-                if (Logger.IsInfoEnabled)
-                    Logger.Info($"Failed to parse {filter} from {path}. Line was {parsedLine}");
-                return 0;
-            }
+                if (parsedLine.Length != 3) // format should be: {filter}: <num> kb
+                {
+                    if (Logger.IsInfoEnabled)
+                        Logger.Info($"Failed to read {filters[cnt]} from {path}. Line was {parsedLine}");
+                    rcValues[cnt++] = 0;
+                    continue;
+                }
 
-            switch (parsedLine[2].ToLowerInvariant())
-            {
-                case "kb":
-                    result *= 1024L;
-                    break;
-                case "mb":
-                    result *= 1024L * 1024;
-                    break;
-                case "gb":
-                    result *= 1024L * 1024 * 1024;
-                    break;
+                var found = false;
+                foreach (var filter in filters)
+                {
+                    if (parsedLine[0].Contains($"{filter}:") == true)
```

Comment:
Is it ok to be case sensitive here?

Reply:
Yes, we can expect it to be stable. But it is called a lot and we don't want to pay all these costs